test/e2e/test_catalogue_category_property.py

```diff
@@ -137,12 +139,29 @@ def setup(self, test_client):
         """Setup fixtures"""
 
         self.test_client = test_client
+        self.units = []
 
     def post_catalogue_category_and_items(self):
         """Posts a catalogue category, catalogue item and item for create tests to act on"""
 
         # pylint:disable=duplicate-code
-        response = self.test_client.post("/v1/catalogue-categories", json=CATALOGUE_CATEGORY_POST_A)
+
+        # units
+        response = self.test_client.post("/v1/units", json=UNIT_POST_A)
+        unit_mm = response.json()
+
+        self.units = [unit_mm]
+
+        response = self.test_client.post(
+            "/v1/catalogue-categories",
+            json={
+                **CATALOGUE_CATEGORY_POST_A,
+                "catalogue_item_properties": add_ids_to_properties(
+                    None, CATALOGUE_CATEGORY_POST_A["catalogue_item_properties"], self.units
+                ),
```

Comment:
I think we need a separate function just for units here actually, it works because the schema will ignore extra parameters but I think for e2e tests its best if we only supply what is needed, and here it will add ids and will still have the unit inside the request, whereas it should effectively replace the unit with unit id and leave the properties alone. I think the unit tests, or where its used in the output check is fine though as they do need both.